[Feature] Add "All" category to template workflows

christian-byrne opened the pull request:
Adds "All" category to template workflows and refactors the template management into a dedicated composable. This PR improves workflow template organization by providing a centralized way to browse all available templates regardless of source, supported by proper test coverage.

Changes to data structure will also make it easier to add a search feature.

┆Issue is synchronized with this [Notion page](https://www.notion.so/PR-3931-Feature-Add-All-category-to-template-workflows-1f76d73d3650811da74bd69dd2dbac81) by [Unito](https://www.unito.io)

christian-byrne committed:
add all category to templates

Commit:
Update locales [skip ci]

christian-byrne committed:
Add useTemplateWorkflows composable for template workflow management

christian-byrne committed:
Move All Templates category into ComfyUI Examples

Commit:
Update locales [skip ci]

christian-byrne committed:
[test] Fix TemplateWorkflowCard test by properly mocking useTemplateWorkflows

christian-byrne commented:
The two failing tests are from the default category now being "All" rather than "Basics"

![Selection_1470](https://github.com/user-attachments/assets/e19f95e1-81b6-4fc6-af6e-1b0111ad93c3)

![Selection_1469](https://github.com/user-attachments/assets/79a1a1aa-c5ac-4570-b9dc-2d920c2d5f73)

Commit:
Update test expectations [skip ci]

christian-byrne commented:
Would like to merge this. Note:

- This was actually part of the original design for templates feature, but considered too much effort to implement at the time
- Will make implementing search feature easy (https://github.com/Comfy-Org/ComfyUI_frontend/issues/3736). (We just auto-focus the All tab when searching.)

Any objection @webfiltered?

webfiltered reviewed (changes requested):
Mostly LGTM - I think the computed changes (last two) should go in.

The unused func nits aren't important though.

christian-byrne committed:
[refactor] optimize template list rendering with computed properties - addresses @webfiltered's performance suggestions

Convert repeated function calls for template title and description to computed properties in TemplateWorkflowList.vue to avoid redundant computations during rendering.

christian-byrne committed:
[refactor] remove unused fetchTemplateJson from composable exports - addresses @webfiltered's API cleanup suggestion

Remove fetchTemplateJson from useTemplateWorkflows exports since it's only used internally within the composable. Update corresponding test mocks.

christian-byrne commented on src/composables/useTemplateWorkflows.ts:
```diff
@@ -0,0 +1,191 @@
+    getTemplateTitle,
+    getTemplateDescription,
+    loadWorkflowTemplate,
+    fetchTemplateJson
```
Removed the unused ones. One of them is used in tests. Should be okay since only calling the constructor `use` function via destructuring so it's always GC.
